### PR TITLE
feat: add synth-switch lambda

### PR DESCRIPTION
**Pull request by ConjunctiveNormalForm**
add `/synthetic-swtich/enabled` endpoint. 

will add some basic handler tests
Screenshot 2023-08-30 at 11 16 14 PM

### CODE REVIEW

File: bin/stacks/api-stack.ts [hunk 1]
```diff
@@ -283,6 +308,16 @@ export class APIStack extends cdk.Stack {
     });
     quote.addMethod('POST', quoteLambdaIntegration);
 
+    const switchLambdaIntegration = new aws_apigateway.LambdaIntegration(switchLambdaAlias, {});
+    const switchResource = api.root.addResource('synthetic-switch', {
+      defaultCorsPreflightOptions: {
+        allowOrigins: aws_apigateway.Cors.ALL_ORIGINS,
+        allowMethods: aws_apigateway.Cors.ALL_METHODS,
```
Comment (zhongeric) [inline bin/stacks/api-stack.ts:315]: we should only allow URA to call this endpoint right? or else people will be able to sniff out whats enabled / not

Comment (zhongeric) [reply bin/stacks/api-stack.ts:315]: not CORS related but we can setup an internal api key
=

Comment (ConjunctiveNormalForm) [reply bin/stacks/api-stack.ts:315]: added auth key and made it required for `enabled` endpoint

File: bin/stacks/api-stack.ts [hunk 2]
```diff
@@ -316,7 +316,18 @@ export class APIStack extends cdk.Stack {
       },
     });
     const enabled = switchResource.addResource('enabled');
-    enabled.addMethod('GET', switchLambdaIntegration);
+
+    /* add auth key */
+    const apiAuthzKey = api.addApiKey('AuthzKey');
```
Comment (zhongeric) [inline bin/stacks/api-stack.ts:321]: where is this value being generated / fetched? I was envisioning a pattern more similar to how we do the GPA internal key for URA

Comment (ConjunctiveNormalForm) [reply bin/stacks/api-stack.ts:321]: key is generated right here! We store it in pipeline secret manager and URA reads it and include it in x-api-key header when calling `enabled`

Comment (zhongeric) [reply bin/stacks/api-stack.ts:321]: oh cool, is Authz a typo or related to this?

File: lib/repositories/switch-repository.ts [hunk 3]
```diff
@@ -48,32 +47,29 @@ export class SwitchRepository implements BaseSwitchRepository {
     private readonly switchEntity: Entity
   ) {}
 
-  public async syntheticQuoteForTradeEnabled(trade: SynthSwitchRequestBody): Promise<boolean> {
+  public async syntheticQuoteForTradeEnabled(trade: SynthSwitchQueryParams): Promise<boolean> {
     const { inputToken, inputTokenChainId, outputToken, outputTokenChainId, type, amount } = trade;
 
+    SwitchRepository.log.info({ trade: trade });
```
Comment (zhongeric) [inline lib/repositories/switch-repository.ts:53]: this.log

**Review by zhongeric — APPROVED** [review]

File: lib/repositories/switch-repository.ts (hunk 3 above)
Comment (ConjunctiveNormalForm) [reply lib/repositories/switch-repository.ts]: log is a static member here

**Pull request merged**